examples/se_2_3_localization.cpp

```diff
@@ -0,0 +1,452 @@
+                             " Vel: " << X_simulation.linearVelocity().transpose() << endl;
+    cout << "X estimated : Pos: " << X.translation().transpose() <<
+                             " RPY: " << X.rotation().eulerAngles(2, 1, 0).reverse().transpose() <<
+                             " Vel: " << X.linearVelocity().transpose() << endl;
+    cout << "X unfilterd : Pos: " << X_unfiltered.translation().transpose() <<
+                             " RPY: " << X_unfiltered.rotation().eulerAngles(2, 1, 0).reverse().transpose() <<
+                             " Vel: " << X_unfiltered.linearVelocity().transpose() << endl;
+    cout << "---------------------------------------------------------------------------" << endl;
+    // END DEBUG
+
+
+    // START TEMPORAL LOOP
+    //
+    //
+
+    // Make 10 steps. Measure up to three landmarks each time.
+    for (double t = 0; t < 10; t+=dt)
+    {
+        //// I. Simulation ###############################################################################
+
+        /// get current simulated state and measurements from previous step
+        auto R_k = X_simulation.rotation();
+        auto v_k = X_simulation.linearVelocity();
+        auto acc_k = alpha_prev + R_k.transpose()*g;
+
+        /// input vector
+        u_nom << (dt*(R_k.transpose())*v_k + 0.5*dt*dt*acc_k),  dt*omega_prev, dt*acc_k;
+
+        /// simulate noise
+        u_noise = u_sigmas * Array9d::Random();             // control noise
+        u_noisy = u_nom + u_noise;                          // noisy control
+
+        u_simu   = u_nom;
+        u_unfilt = u_noisy;
+
+        /// first we move - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+        X_simulation = X_simulation + u_simu;               // overloaded X.rplus(u) = X * exp(u)
+        /// update expected IMU measurements
+        alpha = alpha_const - X_simulation.rotation().transpose()*g; // update expected IMU measurement after moving
+
+        /// then we measure all landmarks - - - - - - - - - - - - - - - - - - - -
+        for (int i = 0; i < landmarks.size(); i++)
+        {
+            b = landmarks[i];                               // lmk coordinates in world frame
+
+            /// simulate noise
+            y_noise = y_sigmas * Array3d::Random();         // measurement noise
+
+            y = X_simulation.inverse().act(b);              // landmark measurement, before adding noise
+            y = y + y_noise;                                // landmark measurement, noisy
+            measurements[i] << y;                     // store for the estimator just below
+        }
+
+
+
+
+        //// II. Estimation ###############################################################################
+
+        /// get current state estimate to build the state-dependent control vector
+        auto R_k_est = X.rotation();
+        auto v_k_est = X.linearVelocity();
+        auto acc_k_est = alpha_prev + R_k_est.transpose()*g;
```

Comment:
Can this be done with X.act() ? Or maybe X.inv().act()?

Reply:
The X.act() will cause the position to be added to this equation.

Reply:
OK